File: soundswap/tools/patch_screen.dart
import 'dart:io';

void main() {
  String content = File('lib/features/overlay_tools/presentation/screens/overlay_tools_screen.dart').readAsStringSync();

  if (!content.contains("import '../widgets/overlay_layers_panel.dart';")) {
    content = content.replaceFirst("import 'package:soundswap/shared/widgets/empty_state.dart';", "import 'package:soundswap/shared/widgets/empty_state.dart';\nimport '../widgets/overlay_layers_panel.dart';\nimport '../widgets/overlay_properties_panel.dart';");
  }

  // Replace _buildLayersTab call
  content = content.replaceAll('_buildLayersTab(context)', 'OverlayLayersPanel(controller: widget.controller)');
  // Replace _buildTransformTab call
  content = content.replaceAll('_buildTransformTab(context)', 'OverlayPropertiesPanel(controller: widget.controller)');

  // Remove _buildLayersTab definition
  int start = content.indexOf('  Widget _buildLayersTab(BuildContext context) {');
  int end = content.indexOf('  Widget _buildTransformTab(BuildContext context) {');
  if (start != -1 && end != -1) {
    content = content.replaceRange(start, end, '');
  }

  // Remove _buildTransformTab definition
  start = content.indexOf('  Widget _buildTransformTab(BuildContext context) {');
  end = content.indexOf('  Widget _buildAlignTab(BuildContext context) {');
  if (start != -1 && end != -1) {
    content = content.replaceRange(start, end, '');
  }

  // Remove _OverlayListTile definition
  start = content.indexOf('class _OverlayListTile extends StatelessWidget {');
  end = content.indexOf('class _TemplateTile extends StatelessWidget {');
  if (start != -1 && end != -1) {
    content = content.replaceRange(start, end, '');
  }

  // Remove controllers and sync state
  start = content.indexOf('  final _nameController = TextEditingController();');
  end = content.indexOf('  @override\n  Widget build(BuildContext context) {');
  if (start != -1 && end != -1) {
    content = content.replaceRange(start, end, '''
  VideoOutputSize _previewSize = VideoOutputSize.vertical1080;
  bool _showGrid = false;
  bool _enableSnapping = true;
  double _zoomScale = 1.0;

''');
  }

  // Remove _buildCleanImagePath
  start = content.indexOf('  Widget _buildCleanImagePath(BuildContext context, String? imagePath) {');
  end = content.indexOf('class _TemplateTile extends StatelessWidget {');
  if (start != -1 && end != -1) {
    content = content.replaceRange(start, end, '');
  }

  File('lib/features/overlay_tools/presentation/screens/overlay_tools_screen.dart').writeAsStringSync(content);
  print('Patched successfully via Dart.');
}

File: soundswap/tools/patch_screen.py
import re

with open('lib/features/overlay_tools/presentation/screens/overlay_tools_screen.dart', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Add imports
imports = "import '../widgets/overlay_layers_panel.dart';\nimport '../widgets/overlay_properties_panel.dart';\n"
for i, line in enumerate(lines):
    if "import 'package:soundswap/shared/widgets/empty_state.dart';" in line:
        lines.insert(i + 1, imports)
        break

with open('lib/features/overlay_tools/presentation/screens/overlay_tools_screen.dart', 'w', encoding='utf-8') as f:
    f.writelines(lines)

with open('lib/features/overlay_tools/presentation/screens/overlay_tools_screen.dart', 'r', encoding='utf-8') as f:
    content = f.read()

# Replace _buildLayersTab call
content = content.replace('_buildLayersTab(context)', 'OverlayLayersPanel(controller: widget.controller)')
# Replace _buildTransformTab call
content = content.replace('_buildTransformTab(context)', 'OverlayPropertiesPanel(controller: widget.controller)')

# Remove _buildLayersTab definition
start = content.find('Widget _buildLayersTab(BuildContext context) {')
end = content.find('Widget _buildTransformTab(BuildContext context) {')
if start != -1 and end != -1:
    content = content[:start] + content[end:]

# Remove _buildTransformTab definition
start = content.find('Widget _buildTransformTab(BuildContext context) {')
end = content.find('Widget _buildAlignTab(BuildContext context) {')
if start != -1 and end != -1:
    content = content[:start] + content[end:]

# Remove _OverlayListTile definition
start = content.find('class _OverlayListTile extends StatelessWidget {')
end = content.find('class _TemplateTile extends StatelessWidget {')
if start != -1 and end != -1:
    content = content[:start] + content[end:]

# Remove controllers and sync state
start = content.find('  final _nameController = TextEditingController();')
end = content.find('  @override\n  Widget build(BuildContext context) {')
if start != -1 and end != -1:
    content = content[:start] + '''
  VideoOutputSize _previewSize = VideoOutputSize.vertical1080;
  bool _showGrid = false;
  bool _enableSnapping = true;
  double _zoomScale = 1.0;

''' + content[end:]

# Remove _buildCleanImagePath
start = content.find('  Widget _buildCleanImagePath(BuildContext context, String? imagePath) {')
end = content.find('class _TemplateTile extends StatelessWidget {')
if start != -1 and end != -1:
    content = content[:start] + content[end:]


with open('lib/features/overlay_tools/presentation/screens/overlay_tools_screen.dart', 'w', encoding='utf-8') as f:
    f.write(content)

print('Patched successfully via Python.')
